xdsl/dialects/arm/ops.py

```diff
@@ -154,3 +156,38 @@ def __init__(
 
     def assembly_line_args(self):
         return (self.d, self.s1, self.s2)
+
+
+@irdl_op_definition
+class RetOp(ARMInstruction):
+    """
+    Return from subroutine.
+
+    Equivalent to `bx lr`
+    """
+
+    name = "arm.return"
+
+    assembly_format = "attr-dict"
+
+    traits = traits_def(IsTerminator())
+
+    def __init__(
+        self,
+        *,
+        comment: str | StringAttr | None = None,
+    ):
+        if isinstance(comment, str):
+            comment = StringAttr(comment)
+
+        super().__init__(
+            attributes={
+                "comment": comment,
+            },
+        )
+
+    def assembly_line_args(self):
+        return ()
+
+    def assembly_instruction_name(self) -> str:
+        return "bx lr"
```

Comment:
this should go in a new `arm_func` dialect, not `arm`, as the name of the op does not match the name of the assembly operation, and neither does the semantics